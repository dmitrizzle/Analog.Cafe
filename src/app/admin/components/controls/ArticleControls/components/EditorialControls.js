import { ButtonStrip } from "@roast-cms/react-button-beans"
import React from "react"

import { TEXT_EMOJIS } from "../../../../../constants"
import ButtonStripItem from "../../../../../core/components/controls/Button/components/ButtonStripItem"

export default props => {
  return (
    <ButtonStrip
      style={{
        margin: "1em auto 0",
        width: "16em",
        whiteSpace: "nowrap",
        display: props.article.status !== "rejected" ? "block" : "none"
      }}
    >
      <div>
        <ButtonStripItem
          left
          onClick={props.edit}
          style={{
            minWidth: "5em"
          }}
        >
          <span role="img" aria-label="(Un)Locked button">
            {props.stateAllowOverwrite
              ? TEXT_EMOJIS.UNLOCKED
              : TEXT_EMOJIS.LOCKED}
          </span>{" "}
          Edit
        </ButtonStripItem>
        {props.article.status === "published" ? (
          <ButtonStripItem
            key="ButtonStrip_Item_unpublish"
            onClick={props.unpublish}
            right
          >
            <span role="img" aria-label="(Un)Locked button">
              {props.stateAllowUnpublish
                ? TEXT_EMOJIS.UNLOCKED
                : TEXT_EMOJIS.LOCKED}
            </span>{" "}
            Unpublish
          </ButtonStripItem>
        ) : (
          [
            <ButtonStripItem
              key="ButtonStrip_Item_reject"
              onClick={props.reject}
              inverse={props.editor.reject.id === props.article.id}
              style={{
                minWidth: "6em",
                display: props.article.status !== "scheduled" ? "block" : "none"
              }}
            >
              <span role="img" aria-label="(Un)Locked button">
                {props.stateAllowReject
                  ? TEXT_EMOJIS.UNLOCKED
                  : TEXT_EMOJIS.LOCKED}
              </span>{" "}
              Reject
            </ButtonStripItem>,
            <ButtonStripItem
              right
              inverse={props.statePublishControls}
              style={
                props.article.status === "scheduled" ? { minWidth: "8em" } : {}
              }
              onClick={
                props.article.status !== "scheduled"
                  ? props.showPublishControls
                  : null
              }
              key="ButtonStrip_Item_publish"
            >
              {props.article.status !== "scheduled"
                ? "Publish"
                : "Edit Schedule"}
            </ButtonStripItem>
          ]
        )}
      </div>
    </ButtonStrip>
  )
}
