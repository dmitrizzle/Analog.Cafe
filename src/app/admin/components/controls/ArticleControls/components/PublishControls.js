import { ButtonStrip } from "@roast-cms/react-button-beans"
import React from "react"

import { TEXT_EMOJIS } from "../../../../../constants"
import Button from "../../../../../core/components/controls/Button/components/Button"
import ButtonStripItem from "../../../../../core/components/controls/Button/components/ButtonStripItem"
import CardIntegrated from "../../../../../core/components/controls/Card/components/CardIntegrated"

const TAGS = {
  culture: "Culture",
  places: "Places",
  "film-photography": "Photography",
  editorials: "Editorials",
  opinions: "Opinions"
}

export default props => {
  return (
    <div style={{ display: props.statePublishControls ? "block" : "none" }}>
      <ButtonStrip
        style={{
          margin: "0.5em auto 0.075em",
          width: "auto",
          overflow: "scroll",
          padding: "1px 1px 3px"
        }}
      >
        <div style={{ minWidth: "30em" }}>
          {Object.keys(TAGS).map((key, i) => {
            let last = i === Object.keys(TAGS).length - 1
            return (
              <ButtonStripItem
                left={i === 0}
                right={last}
                key={key}
                onClick={event => props.setPublicationTag(event, key)}
                style={key === "film-photography" ? { minWidth: "7.5em" } : {}}
                inverse={props.statePublishAs === key}
                to={`#${key}`}
              >
                {TAGS[key]}
              </ButtonStripItem>
            )
          })}
        </div>
      </ButtonStrip>
      <CardIntegrated
        style={{
          marginBottom: 0,
          marginTop: "0.5em",
          display: props.statePublishAs ? "block" : "none"
        }}
      >
        <Button branded>Add to Queue</Button>
        <Button
          onClick={props.publishNow}
          loading={props.editor.publish.id === props.article.id}
        >
          {props.editor.publish.id !== props.article.id &&
            (props.stateAllowPublish
              ? TEXT_EMOJIS.UNLOCKED
              : TEXT_EMOJIS.LOCKED)}{" "}
          Publish Now
        </Button>
      </CardIntegrated>
    </div>
  )
}
