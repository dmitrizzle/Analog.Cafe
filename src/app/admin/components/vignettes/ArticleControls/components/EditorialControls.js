import React from "react"

import {
  ButtonStrip,
  Item
} from "../../../../../core/components/controls/ButtonStrip"
import { TEXT_EMOJIS } from "../../../../../constants"

export default props => {
  return (
    <ButtonStrip
      style={{
        margin: "1em auto 0",
        width: "16em",
        display: props.article.status !== "rejected" ? "block" : "none"
      }}
    >
      <div>
        <Item
          left
          onClick={props.edit}
          style={{
            minWidth: "5em"
          }}
        >
          <span role="img" aria-label="(Un)Locked button">
            {props.stateAllowOverwrite
              ? TEXT_EMOJIS.LOCKED
              : TEXT_EMOJIS.UNLOCKED}
          </span>{" "}
          Edit
        </Item>
        {props.article.status === "published"
          ? [
              <Item key="ButtonStrip_Item_update" onClick={props.sync}>
                <span role="img" aria-label="(Un)Locked button">
                  {props.stateAllowSync
                    ? TEXT_EMOJIS.LOCKED
                    : TEXT_EMOJIS.UNLOCKED}
                </span>{" "}
                Sync
              </Item>,
              <Item key="ButtonStrip_Item_unpublish" right>
                Unpublish
              </Item>
            ]
          : [
              <Item
                key="ButtonStrip_Item_reject"
                onClick={props.reject}
                inverse={props.editor.reject.id === props.article.id}
                style={{
                  minWidth: "6em",
                  display:
                    props.article.status !== "scheduled" ? "block" : "none"
                }}
              >
                <span role="img" aria-label="(Un)Locked button">
                  {props.stateAllowReject
                    ? TEXT_EMOJIS.LOCKED
                    : TEXT_EMOJIS.UNLOCKED}
                </span>{" "}
                Reject
              </Item>,
              <Item
                right
                inverse={props.statePublishControls}
                style={
                  props.article.status === "scheduled"
                    ? { minWidth: "8em" }
                    : {}
                }
                onClick={
                  props.article.status !== "scheduled"
                    ? props.showPublishControls
                    : null
                }
                key="ButtonStrip_Item_publish"
              >
                {props.article.status !== "scheduled"
                  ? "Publish"
                  : "Edit Schedule"}
              </Item>
            ]}
      </div>
    </ButtonStrip>
  )
}
