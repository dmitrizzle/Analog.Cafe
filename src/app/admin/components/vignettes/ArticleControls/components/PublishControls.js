import React from "react"

import { Button } from "../../../../../core/components/controls/Button"
import {
  ButtonStrip,
  Item
} from "../../../../../core/components/controls/ButtonStrip"
import { CardFlattened } from "../../../../../core/components/controls/Card/styles"
import { TEXT_EMOJIS } from "../../../../../constants"

const TAGS = {
  story: "Story",
  editorial: "Editorial",
  guide: "Guide",
  review: "Review",
  "photo-essay": "Photo Essay"
}

export default props => {
  return (
    <div style={{ display: props.statePublishControls ? "block" : "none" }}>
      <ButtonStrip
        style={{
          margin: "0.5em auto 0",
          width: "auto",
          overflow: "scroll",
          padding: "1px 1px 3px"
        }}
      >
        <div>
          {Object.keys(TAGS).map((key, i) => {
            let last = i === Object.keys(TAGS).length - 1
            return (
              <Item
                left={i === 0}
                right={last}
                key={key}
                onClick={event => props.setPublicationTag(event, key)}
                style={key === "photo-essay" ? { minWidth: "7.5em" } : {}}
                inverse={props.statePublishAs === key}
                to={`#${key}`}
              >
                {TAGS[key]}
              </Item>
            )
          })}
        </div>
      </ButtonStrip>
      <CardFlattened
        style={{
          marginBottom: 0,
          marginTop: "0.5em",
          display: props.statePublishAs ? "block" : "none"
        }}
      >
        <Button branded>Add to Queue</Button>
        <Button
          onClick={props.publishNow}
          loading={props.editor.publish.id === props.article.id}
        >
          {props.editor.publish.id !== props.article.id &&
            (props.stateAllowPublish
              ? TEXT_EMOJIS.LOCKED
              : TEXT_EMOJIS.UNLOCKED)}{" "}
          Publish Now
        </Button>
      </CardFlattened>
    </div>
  )
}
