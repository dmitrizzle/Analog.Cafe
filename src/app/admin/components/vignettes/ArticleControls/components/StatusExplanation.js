import React from "react"

import {
  ROUTE_URL_ARTICLES,
  ROUTE_URL_SUBMISSIONS
} from "../../../../../core/constants/routes-article"
import { TEXT_EMOJIS } from "../../../../../constants"
import Byline from "../../../../../core/components/vignettes/Byline"
import Link from "../../../../../core/components/controls/Link"

export default props => {
  return (
    <Byline
      style={{
        display: "block"
      }}
    >
      {(props.article.status === "scheduled" ||
        props.article.status === "published") && (
        <span style={{ fontStyle: "normal" }} role="img" aria-label="Notice">
          {TEXT_EMOJIS.WARNING}
        </span>
      )}{" "}
      {props.article.status === "scheduled" &&
        "This is a SCHEDULED submission that you can edit while it is in the queue."}
      {props.article.status === "published" &&
        props.article.articleId && (
          <span>
            This is an <strong>original submission</strong>, linked with{" "}
            <Link to={`${ROUTE_URL_ARTICLES}/${props.article.slug}`}>
              this published article
            </Link>. You can edit it and then sync your changes to the published
            article.
          </span>
        )}
      {props.article.status === "published" &&
        !props.article.articleId && (
          <span>
            This is a <strong>published article</strong>, linked with{" "}
            <Link to={`${ROUTE_URL_SUBMISSIONS}/${props.article.slug}`}>
              this submission
            </Link>. Your edits will be applied to that submission, which you
            can then sync with this article.
          </span>
        )}
    </Byline>
  )
}
