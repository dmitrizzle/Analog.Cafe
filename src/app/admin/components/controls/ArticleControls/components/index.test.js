import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import EditorialControls from "./EditorialControls"
import PublishControls from "./PublishControls"
import StatusExplanation from "./StatusExplanation"

it("Render PublishControls without crashing, matches snapshot", () => {
  const element = shallow(
    <PublishControls
      editor={{
        publish: {
          id: 0
        }
      }}
      article={{
        id: 0
      }}
    />
  )
  expect(element).toMatchSnapshot()
})

it("Render EditorialControls without crashing, matches snapshot", () => {
  const element = shallow(
    <EditorialControls
      article={{
        status: "published"
      }}
    />
  )
  expect(element).toMatchSnapshot()
})

it("Render StatusExplanation without crashing, matches snapshot", () => {
  const element = shallow(
    <StatusExplanation article={{ status: "scheduled" }} />
  )
  expect(element).toMatchSnapshot()
})
