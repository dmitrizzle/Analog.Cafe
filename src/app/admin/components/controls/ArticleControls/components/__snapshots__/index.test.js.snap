// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render EditorialControls without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    article={
        Object {
            "status": "published",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Unknown
                left={true}
                onClick={undefined}
                style={
                        Object {
                                "minWidth": "5em",
                              }
                }
        >
                <span
                        aria-label="(Un)Locked button"
                        role="img"
                >
                        🔐
                </span>
                 
                Edit
        </Unknown>
        <Unknown
                onClick={undefined}
                right={true}
        >
                <span
                        aria-label="(Un)Locked button"
                        role="img"
                >
                        🔐
                </span>
                 
                Unpublish
        </Unknown>
</div>,
      "style": Object {
        "display": "block",
        "margin": "1em auto 0",
        "whiteSpace": "nowrap",
        "width": "16em",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            left={true}
            onClick={undefined}
            style={
                        Object {
                                    "minWidth": "5em",
                                  }
            }
>
            <span
                        aria-label="(Un)Locked button"
                        role="img"
            >
                        🔐
            </span>
             
            Edit
</Unknown>,
          <Unknown
            onClick={undefined}
            right={true}
>
            <span
                        aria-label="(Un)Locked button"
                        role="img"
            >
                        🔐
            </span>
             
            Unpublish
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span
                aria-label="(Un)Locked button"
                role="img"
>
                🔐
</span>,
              " ",
              "Edit",
            ],
            "left": true,
            "onClick": undefined,
            "style": Object {
              "minWidth": "5em",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "(Un)Locked button",
                "children": "🔐",
                "role": "img",
              },
              "ref": null,
              "rendered": "🔐",
              "type": "span",
            },
            " ",
            "Edit",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ButtonStrip_Item_unpublish",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span
                aria-label="(Un)Locked button"
                role="img"
>
                🔐
</span>,
              " ",
              "Unpublish",
            ],
            "onClick": undefined,
            "right": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "(Un)Locked button",
                "children": "🔐",
                "role": "img",
              },
              "ref": null,
              "rendered": "🔐",
              "type": "span",
            },
            " ",
            "Unpublish",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Unknown
                    left={true}
                    onClick={undefined}
                    style={
                              Object {
                                        "minWidth": "5em",
                                      }
                    }
          >
                    <span
                              aria-label="(Un)Locked button"
                              role="img"
                    >
                              🔐
                    </span>
                     
                    Edit
          </Unknown>
          <Unknown
                    onClick={undefined}
                    right={true}
          >
                    <span
                              aria-label="(Un)Locked button"
                              role="img"
                    >
                              🔐
                    </span>
                     
                    Unpublish
          </Unknown>
</div>,
        "style": Object {
          "display": "block",
          "margin": "1em auto 0",
          "whiteSpace": "nowrap",
          "width": "16em",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              left={true}
              onClick={undefined}
              style={
                            Object {
                                          "minWidth": "5em",
                                        }
              }
>
              <span
                            aria-label="(Un)Locked button"
                            role="img"
              >
                            🔐
              </span>
               
              Edit
</Unknown>,
            <Unknown
              onClick={undefined}
              right={true}
>
              <span
                            aria-label="(Un)Locked button"
                            role="img"
              >
                            🔐
              </span>
               
              Unpublish
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span
                  aria-label="(Un)Locked button"
                  role="img"
>
                  🔐
</span>,
                " ",
                "Edit",
              ],
              "left": true,
              "onClick": undefined,
              "style": Object {
                "minWidth": "5em",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "(Un)Locked button",
                  "children": "🔐",
                  "role": "img",
                },
                "ref": null,
                "rendered": "🔐",
                "type": "span",
              },
              " ",
              "Edit",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ButtonStrip_Item_unpublish",
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span
                  aria-label="(Un)Locked button"
                  role="img"
>
                  🔐
</span>,
                " ",
                "Unpublish",
              ],
              "onClick": undefined,
              "right": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "(Un)Locked button",
                  "children": "🔐",
                  "role": "img",
                },
                "ref": null,
                "rendered": "🔐",
                "type": "span",
              },
              " ",
              "Unpublish",
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render PublishControls without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    article={
        Object {
            "id": 0,
          }
    }
    editor={
        Object {
            "publish": Object {
              "id": 0,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.div
          style={
                    Object {
                              "margin": "0.5em auto 0.075em",
                              "overflow": "scroll",
                              "padding": "1px 1px 3px",
                              "width": "auto",
                            }
          }
>
          <div
                    style={
                              Object {
                                        "margin": "0 auto",
                                        "width": "26em",
                                      }
                    }
          >
                    <Unknown
                              inverse={false}
                              left={true}
                              onClick={[Function]}
                              right={false}
                              style={Object {}}
                              to="#photo-essay"
                    >
                              Photo Essay
                    </Unknown>
                    <Unknown
                              inverse={false}
                              left={false}
                              onClick={[Function]}
                              right={false}
                              style={
                                        Object {
                                                  "minWidth": "10.5em",
                                                }
                              }
                              to="#film-photography"
                    >
                              Film Photography
                    </Unknown>
                    <Unknown
                              inverse={false}
                              left={false}
                              onClick={[Function]}
                              right={true}
                              style={Object {}}
                              to="#editorial"
                    >
                              Editorial
                    </Unknown>
          </div>
</styled.div>,
        <Styled(styled.div)
          style={
                    Object {
                              "display": "none",
                              "marginBottom": 0,
                              "marginTop": "0.5em",
                            }
          }
>
          <Unknown
                    branded={true}
          >
                    Add to Queue
          </Unknown>
          <Unknown
                    loading={true}
                    onClick={undefined}
          >
                     
                    Publish Now
          </Unknown>
</Styled(styled.div)>,
      ],
      "style": Object {
        "display": "none",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            style={
                        Object {
                                    "margin": "0 auto",
                                    "width": "26em",
                                  }
            }
>
            <Unknown
                        inverse={false}
                        left={true}
                        onClick={[Function]}
                        right={false}
                        style={Object {}}
                        to="#photo-essay"
            >
                        Photo Essay
            </Unknown>
            <Unknown
                        inverse={false}
                        left={false}
                        onClick={[Function]}
                        right={false}
                        style={
                                    Object {
                                                "minWidth": "10.5em",
                                              }
                        }
                        to="#film-photography"
            >
                        Film Photography
            </Unknown>
            <Unknown
                        inverse={false}
                        left={false}
                        onClick={[Function]}
                        right={true}
                        style={Object {}}
                        to="#editorial"
            >
                        Editorial
            </Unknown>
</div>,
          "style": Object {
            "margin": "0.5em auto 0.075em",
            "overflow": "scroll",
            "padding": "1px 1px 3px",
            "width": "auto",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                inverse={false}
                left={true}
                onClick={[Function]}
                right={false}
                style={Object {}}
                to="#photo-essay"
>
                Photo Essay
</Unknown>,
              <Unknown
                inverse={false}
                left={false}
                onClick={[Function]}
                right={false}
                style={
                                Object {
                                                "minWidth": "10.5em",
                                              }
                }
                to="#film-photography"
>
                Film Photography
</Unknown>,
              <Unknown
                inverse={false}
                left={false}
                onClick={[Function]}
                right={true}
                style={Object {}}
                to="#editorial"
>
                Editorial
</Unknown>,
            ],
            "style": Object {
              "margin": "0 auto",
              "width": "26em",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "photo-essay",
              "nodeType": "function",
              "props": Object {
                "children": "Photo Essay",
                "inverse": false,
                "left": true,
                "onClick": [Function],
                "right": false,
                "style": Object {},
                "to": "#photo-essay",
              },
              "ref": null,
              "rendered": "Photo Essay",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "film-photography",
              "nodeType": "function",
              "props": Object {
                "children": "Film Photography",
                "inverse": false,
                "left": false,
                "onClick": [Function],
                "right": false,
                "style": Object {
                  "minWidth": "10.5em",
                },
                "to": "#film-photography",
              },
              "ref": null,
              "rendered": "Film Photography",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "editorial",
              "nodeType": "function",
              "props": Object {
                "children": "Editorial",
                "inverse": false,
                "left": false,
                "onClick": [Function],
                "right": true,
                "style": Object {},
                "to": "#editorial",
              },
              "ref": null,
              "rendered": "Editorial",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Unknown
              branded={true}
>
              Add to Queue
</Unknown>,
            <Unknown
              loading={true}
              onClick={undefined}
>
               
              Publish Now
</Unknown>,
          ],
          "style": Object {
            "display": "none",
            "marginBottom": 0,
            "marginTop": "0.5em",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "branded": true,
              "children": "Add to Queue",
            },
            "ref": null,
            "rendered": "Add to Queue",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                false,
                " ",
                "Publish Now",
              ],
              "loading": true,
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Array [
              false,
              " ",
              "Publish Now",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.div
            style={
                        Object {
                                    "margin": "0.5em auto 0.075em",
                                    "overflow": "scroll",
                                    "padding": "1px 1px 3px",
                                    "width": "auto",
                                  }
            }
>
            <div
                        style={
                                    Object {
                                                "margin": "0 auto",
                                                "width": "26em",
                                              }
                        }
            >
                        <Unknown
                                    inverse={false}
                                    left={true}
                                    onClick={[Function]}
                                    right={false}
                                    style={Object {}}
                                    to="#photo-essay"
                        >
                                    Photo Essay
                        </Unknown>
                        <Unknown
                                    inverse={false}
                                    left={false}
                                    onClick={[Function]}
                                    right={false}
                                    style={
                                                Object {
                                                            "minWidth": "10.5em",
                                                          }
                                    }
                                    to="#film-photography"
                        >
                                    Film Photography
                        </Unknown>
                        <Unknown
                                    inverse={false}
                                    left={false}
                                    onClick={[Function]}
                                    right={true}
                                    style={Object {}}
                                    to="#editorial"
                        >
                                    Editorial
                        </Unknown>
            </div>
</styled.div>,
          <Styled(styled.div)
            style={
                        Object {
                                    "display": "none",
                                    "marginBottom": 0,
                                    "marginTop": "0.5em",
                                  }
            }
>
            <Unknown
                        branded={true}
            >
                        Add to Queue
            </Unknown>
            <Unknown
                        loading={true}
                        onClick={undefined}
            >
                         
                        Publish Now
            </Unknown>
</Styled(styled.div)>,
        ],
        "style": Object {
          "display": "none",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              style={
                            Object {
                                          "margin": "0 auto",
                                          "width": "26em",
                                        }
              }
>
              <Unknown
                            inverse={false}
                            left={true}
                            onClick={[Function]}
                            right={false}
                            style={Object {}}
                            to="#photo-essay"
              >
                            Photo Essay
              </Unknown>
              <Unknown
                            inverse={false}
                            left={false}
                            onClick={[Function]}
                            right={false}
                            style={
                                          Object {
                                                        "minWidth": "10.5em",
                                                      }
                            }
                            to="#film-photography"
              >
                            Film Photography
              </Unknown>
              <Unknown
                            inverse={false}
                            left={false}
                            onClick={[Function]}
                            right={true}
                            style={Object {}}
                            to="#editorial"
              >
                            Editorial
              </Unknown>
</div>,
            "style": Object {
              "margin": "0.5em auto 0.075em",
              "overflow": "scroll",
              "padding": "1px 1px 3px",
              "width": "auto",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Unknown
                  inverse={false}
                  left={true}
                  onClick={[Function]}
                  right={false}
                  style={Object {}}
                  to="#photo-essay"
>
                  Photo Essay
</Unknown>,
                <Unknown
                  inverse={false}
                  left={false}
                  onClick={[Function]}
                  right={false}
                  style={
                                    Object {
                                                      "minWidth": "10.5em",
                                                    }
                  }
                  to="#film-photography"
>
                  Film Photography
</Unknown>,
                <Unknown
                  inverse={false}
                  left={false}
                  onClick={[Function]}
                  right={true}
                  style={Object {}}
                  to="#editorial"
>
                  Editorial
</Unknown>,
              ],
              "style": Object {
                "margin": "0 auto",
                "width": "26em",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "photo-essay",
                "nodeType": "function",
                "props": Object {
                  "children": "Photo Essay",
                  "inverse": false,
                  "left": true,
                  "onClick": [Function],
                  "right": false,
                  "style": Object {},
                  "to": "#photo-essay",
                },
                "ref": null,
                "rendered": "Photo Essay",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "film-photography",
                "nodeType": "function",
                "props": Object {
                  "children": "Film Photography",
                  "inverse": false,
                  "left": false,
                  "onClick": [Function],
                  "right": false,
                  "style": Object {
                    "minWidth": "10.5em",
                  },
                  "to": "#film-photography",
                },
                "ref": null,
                "rendered": "Film Photography",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "editorial",
                "nodeType": "function",
                "props": Object {
                  "children": "Editorial",
                  "inverse": false,
                  "left": false,
                  "onClick": [Function],
                  "right": true,
                  "style": Object {},
                  "to": "#editorial",
                },
                "ref": null,
                "rendered": "Editorial",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown
                branded={true}
>
                Add to Queue
</Unknown>,
              <Unknown
                loading={true}
                onClick={undefined}
>
                 
                Publish Now
</Unknown>,
            ],
            "style": Object {
              "display": "none",
              "marginBottom": 0,
              "marginTop": "0.5em",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "branded": true,
                "children": "Add to Queue",
              },
              "ref": null,
              "rendered": "Add to Queue",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  false,
                  " ",
                  "Publish Now",
                ],
                "loading": true,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Array [
                false,
                " ",
                "Publish Now",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render StatusExplanation without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    article={
        Object {
            "status": "scheduled",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span
          aria-label="Notice"
          role="img"
          style={
                    Object {
                              "fontStyle": "normal",
                            }
          }
>
          ⚠️
</span>,
        " ",
        "This is a SCHEDULED submission that you can edit while it is in the queue.",
        false,
        false,
      ],
      "style": Object {
        "display": "block",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Notice",
          "children": "⚠️",
          "role": "img",
          "style": Object {
            "fontStyle": "normal",
          },
        },
        "ref": null,
        "rendered": "⚠️",
        "type": "span",
      },
      " ",
      "This is a SCHEDULED submission that you can edit while it is in the queue.",
      false,
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span
            aria-label="Notice"
            role="img"
            style={
                        Object {
                                    "fontStyle": "normal",
                                  }
            }
>
            ⚠️
</span>,
          " ",
          "This is a SCHEDULED submission that you can edit while it is in the queue.",
          false,
          false,
        ],
        "style": Object {
          "display": "block",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Notice",
            "children": "⚠️",
            "role": "img",
            "style": Object {
              "fontStyle": "normal",
            },
          },
          "ref": null,
          "rendered": "⚠️",
          "type": "span",
        },
        " ",
        "This is a SCHEDULED submission that you can edit while it is in the queue.",
        false,
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
